### PR TITLE
INGK-859 Use latest PySOEM version

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Use the new way of EMCY messages handling from the PySOEM library.

Fixes # INGK-859

### Type of change

Please add a description and delete options that are not relevant.

- Remove retries from the read/write functions.
- Add a method to get the emergency error description.

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

Please describe the tests that you ran to verify your changes if it applies. 

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingenialink tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def get_docs_url():
         "python-can==4.3.1",
         "ingenialogger>=0.2.1",
         "ping3==4.0.3",
-        "pysoem>=1.1.6, <1.2.0",
+        "pysoem@ git+https://github.com/bnjmnp/pysoem.git#master",
```
Comment (pzinemanas) [inline setup.py:47]: Isn't this released yet? If we install pysoem like this, SOEM would not be compiled, right?

Comment (MartinALL91) [reply setup.py:47]: You're right, we are missing the build tools in the Jenkins agents.

File: tests/ethercat/test_ethercat_servo.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    servo, _ = connect_to_slave
+    # Mocker does not detect the calls to ingenialink.ethercat.network.EthercatNetwork._emcy_callback
+    # so we use this function instead.
+    mocked = mocker.patch("ingenialink.ethercat.servo.EthercatServo.get_emergency_description")
+    prev_val = servo.read("DRV_PROT_USER_OVER_VOLT", subnode=1)
+    servo.write("DRV_PROT_USER_OVER_VOLT", data=10.0, subnode=1)
+    with pytest.raises(ILStateError):
+        servo.enable()
+    servo.fault_reset()
+    assert mocked.call_count == 2
```
Comment (polfeliu) [inline tests/ethercat/test_ethercat_servo.py:17]: Why is it called two times? Does the mocker callback record the arguments (description) it was called with?

Comment (MartinALL91) [reply tests/ethercat/test_ethercat_servo.py:17]: The drive sends two emergency messages, one when the fault occurs and one when the fault is cleared (fault reset).

**Review by polfeliu — APPROVED** [review]

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**